src/parent/reducers/reducers.js

```diff
@@ -7,7 +7,7 @@ function parentReducer(state = {}, action) {
 
     case ACTION_TYPES.CLOSE: {
         const newState = Object.assign({}, state);
-        delete newState[action.windowName];
+        delete newState.childWindows[action.windowName];
```

Comment:
I'd consider using `combineReducers`, so you don't have to keep repeating `.childWindows` in this reducer (so it will only receive the state it is responsible for, not the entire state).